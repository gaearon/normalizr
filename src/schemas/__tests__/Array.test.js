// eslint-env jest
import { fromJS } from 'immutable';
import { denormalize, schema } from '../..';

describe(`${schema.Array.name} denormalization`, () => {
  describe('Class', () => {
    test('denormalizes multiple entities with mismatched structures', () => {
      const catSchema = new schema.Entity('cats');
      const peopleSchema = new schema.Entity('person');
      const listSchema = new schema.Array(
        {
          cats: catSchema,
          dogs: {},
          people: peopleSchema,
        },
        (input, parent, key) => input.type || 'dogs'
      );

      const entities = {
        cats: {
          123: {
            id: '123',
            type: 'cats',
          },
          456: {
            id: '456',
            type: 'cats',
          },
        },
        person: {
          123: {
            id: '123',
            type: 'people',
          },
        },
      };

      const input = [
        { id: '123', schema: 'cats' },
        { id: '123', schema: 'people' },
        { id: { id: '789' }, schema: 'dogs' },
        { id: '456', schema: 'cats' },
      ];

      expect(denormalize(input, listSchema, entities)).toMatchSnapshot();
      expect(denormalize(input, listSchema, fromJS(entities))).toMatchSnapshot();
    });
  });
});
