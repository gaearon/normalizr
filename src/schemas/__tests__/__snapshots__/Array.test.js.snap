// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArraySchema denormalization Class denormalizes multiple entities with mismatched structures 1`] = `
Array [
  Object {
    "id": "123",
    "type": "cats",
  },
  Object {
    "id": "123",
    "type": "people",
  },
  Object {
    "id": "789",
  },
  Object {
    "id": "456",
    "type": "cats",
  },
]
`;

exports[`ArraySchema denormalization Class denormalizes multiple entities with mismatched structures 2`] = `
Array [
  Immutable.Map {
    "id": "123",
    "type": "cats",
  },
  Immutable.Map {
    "id": "123",
    "type": "people",
  },
  Object {
    "id": "789",
  },
  Immutable.Map {
    "id": "456",
    "type": "cats",
  },
]
`;
